wsgi/prz-recipe-options.rst

```diff
@@ -23,30 +18,131 @@ The default is to listen on 0.0.0.0:8080.
+For WSGI the default is ``INFO`` (``WARN`` for ZServer).
+
+Advanced logging for WSGI
+-------------------------
+
+``access-log-handler``, ``access-log-args``, ``access-log-kwargs``, ``event-log-handler``, ``event-log-args``, ``event-log-kwargs`` are WSGI only options that you can use to configure arbitrary Python logging handlers.
+Numerous logging handlers are defined in the `Python standard library <https://docs.python.org/3/library/logging.handlers.html>`_.
+
+Exercise 1
+++++++++++
+
+Configure Plone access logging to a TCP Server using ``logging.handlers.SocketHandler`` from the Python standard library.
+Use a local address.
+
+
+..  admonition:: Solution
+    :class: toggle
+
+    Add the following two lines to your buildout configuration (we use ``options.cfg``) from the ``wsgitraining_buildout`` as a starting point:
+
+    .. code-block:: ini
+        :emphasize-lines: 9-10
+
+        ...
+        [instance]
+        recipe = plone.recipe.zope2instance
+        user = admin:admin
+        zeo-client = on
+        zeo-address = 8100
+        shared-blob = on
+        blob-storage = ${buildout:directory}/var/blobstorage
+        access-log-handler = logging.handlers.SocketHandler
+        access-log-args = ('localhost', 9000)
+        eggs =
+            Plone
+            Pillow
+            wsgitraining.site
+
+    ``plone.recipe.zope2instance`` uses default values for the ``access-log-args`` option, but not the ``access-log-kwargs`` option.
+    Same for the ``event-log-args/event-log-kwargs`` options.
+    This means you **have** to provide the ``...-log-args`` parameter, otherwise you will end up with the (in our case nonsensical) defaults in your ``wsgi.ini``.
+    After running buildout with ``buildout -c options.cfg`` you can start your instance with ``bin/instance fg``.
+    Use a tool like e.g. `netcat <http://netcat.sourceforge.net/>`_ (there is a package for your linux distribution) to open a listening socket: ``nc -l 9000``.
+    You will see the incoming log entries in pickled format when navigating to your Plone instance in the browser.
+
+sentry... options
+-----------------
+
+Sentry support for WSGI is  available through ``plone.recipe.zope2instance``.
+We'll cover these options later in the add-ons chapter.
 
 Options that are currently unavailable for WSGI
 -----------------------------------------------
 
-ip-address, ftp-address, icp-address, webdav-address/webdav-force-connection-close, http-header-max-length.
+The following options are currently not available for WSGI:
 
-asyncio vs async - tbd
-----------------------
+* ``access-log-custom``, ``access-log-oldfiles``, ``access-log-max-size``, ``event-log-custom``, ``event-log-oldfiles`` and ``event-log-max-sie`` can be replaced by the new ``...-log-handler``, ``...-log-args`` and ``...-log-kwargs`` options, see above and also the examples given in the recipe `README <https://github.com/plone/plone.recipe.zope2instance#advanced-logging-options-for-wsgi>`_.
```

Comment:
Two sentences, break at "options, see", to become "options. See" with the second sentence on a separate line.